presto-docs/src/main/sphinx/admin/resource-groups.rst

```diff
@@ -9,287 +9,433 @@ Except for the limit on queued queries, when a resource group runs out of a reso
+
+``user_regex``: A regex pattern that matches the user names for this selector.
+
+``source_regex``: A regex pattern that matches the source names for this selector.
+
+Resource Group Properties
+-------------------------
+Resource Groups are defined using a set of properties that determine how resources are allocated and used.
+Here are the key properties that can be set for a Resource Group
+
+* ``name`` (required): The name of the resource group. This is a mandatory property.
+
+* ``maxQueued`` (required): The maximum number of queries that can be queued in the resource group.
+  If this limit is reached, new queries will be rejected.
 
 * ``hardCpuLimit`` (optional): maximum amount of CPU time this
   group may use in a period.
 
-* ``schedulingPolicy`` (optional): specifies how queued queries are selected to run,
-  and how sub-groups become eligible to start their queries. May be one of three values:
+* ``softMemoryLimit`` (required): The maximum amount of memory that the resource group can use.
+  This can be specified in absolute terms (like "10GB") or as a percentage of the total available
+  memory (like "50%").
 
-    * ``fair`` (default): queued queries are processed first-in-first-out, and sub-groups
-      must take turns starting new queries (if they have any queued).
+* ``hardConcurrencyLimit`` (required): The maximum number of queries that the resource group
+  can run concurrently.
 
-    * ``weighted_fair``: sub-groups are selected based on their ``schedulingWeight`` and the number of
-      queries they are already running concurrently. The expected share of running queries for a
-      sub-group is computed based on the weights for all currently eligible sub-groups. The sub-group
-      with the least concurrency relative to its share is selected to start the next query.
+* ``softConcurrencyLimit`` (optional): The soft limit on the number of concurrent queries.
+  If this limit is exceeded, the scheduler will try to prevent new queries from starting,
+  but it won't force running queries to stop.
 
-    * ``weighted``: queued queries are selected stochastically in proportion to their priority
-      (specified via the ``query_priority`` :doc:`session property </sql/set-session>`). Sub groups are selected
-      to start new queries in proportion to their ``schedulingWeight``.
+* ``softCpuLimit`` (optional): maximum amount of CPU time this
+  group may use in a period (see ``cpuQuotaPeriod``), before a penalty is applied to
+  the maximum number of running queries. ``hardCpuLimit`` must also be specified.
 
-    * ``query_priority``: all sub-groups must also be configured with ``query_priority``.
-      Queued queries will be selected strictly according to their priority.
+* ``schedulingPolicy`` (optional): The policy that determines how queries are scheduled within the resource group.
+  This can be set to ``fair``, ``weighted``, ``weighted_fair`` or ``query_priority``. May be one of four values:
```

Comment:
How about rephrasing it to - This can be set to one of the four values ``fair``, ``weighted``, ``weighted_fair`` or ``query_priority`` :